drv/lpc55-update-server/src/main.rs

```diff
@@ -197,6 +225,44 @@ impl idl::InOrderUpdateImpl for ServerImpl {
     }
 }
 
+// Perform some sanity checking on the header block.
+fn validate_header_block(
+    target: UpdateTarget,
+    block: &[u8; BLOCK_SIZE_BYTES],
+) -> Result<(), UpdateError> {
+    let reset_vector = u32::from_le_bytes(
+        block[RESET_VECTOR_OFFSET..][..4].try_into().unwrap_lite(),
```

Comment:
nit: 4 is obvious and understandable and keeps the line short, but I don't like to see constants other than -1, 0, 1, 2 in code and sometimes not those either.

core::mem::size_of::<u32>()